src/main/java/org/maxgamer/quickshop/shop/ShopControlPanel.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    /**
+     * Unregister a registered executor from registry
+     *
+     * @param namespace The plugin instance
+     * @param priority  The entry priority
+     * @param executor  The executor, should returns an QuickComponent or null if no out-put needs
+     */
+    public void unregister(@NotNull Plugin namespace, int priority, @NotNull Function<Map.Entry<@NotNull CommandSender, @NotNull Shop>, @Nullable QuickComponent> executor) {
+        TreeSet<Entry> entries = this.panelRegistry.get(namespace);
+        if (entries == null) {
+            return;
+        }
+        Entry entry = new Entry(priority, executor);
+        entries.remove(entry);
+    }
+
+    /**
+     * Unregister a registered executor from registry
+     *
+     * @param namespace The plugin instance
+     * @param executor  The executor, should returns an QuickComponent or null if no out-put needs
+     */
+    public void unregister(@NotNull Plugin namespace, @NotNull Function<Map.Entry<@NotNull CommandSender, @NotNull Shop>, @Nullable QuickComponent> executor) {
+        TreeSet<Entry> entries = this.panelRegistry.get(namespace);
+        if (entries == null) {
+            return;
+        }
+        entries.removeIf(entry -> entry.getExecutor().equals(executor));
+    }
+
+    /**
+     * Unregister plugin's all registered executor from registry
+     *
+     * @param namespace The plugin instance
+     */
+    public void unregisterAll(@NotNull Plugin namespace) {
+        this.panelRegistry.remove(namespace);
+    }
+
+
+    public static class Entry {
+        private final int priority;
+        private final Function<Map.Entry<@NotNull CommandSender, @NotNull Shop>, @Nullable QuickComponent> executor;
+
+        public Entry(int priority, @NotNull Function<Map.Entry<@NotNull CommandSender, @NotNull Shop>, @Nullable QuickComponent> executor) {
+            this.priority = priority;
+            this.executor = executor;
+        }
+
+        /**
+         * Gets this entry priority
+         *
+         * @return The priority
+         */
+        public int getPriority() {
+            return priority;
+        }
+
+        /**
+         * Gets the executor
+         *
+         * @return The executor
+         */
+        @NotNull
+        public Function<Map.Entry<@NotNull CommandSender, @NotNull Shop>, @Nullable QuickComponent> getExecutor() {
+            return executor;
+        }
+
+        /**
+         * Execute the executor and gets an QuickComponent (or null for no out-put)
+         *
+         * @param data The input data that is an entry contains sender and shop object
+         * @return The QuickComponent or null for no out-put
+         */
+        @Nullable
+        public QuickComponent get(@NotNull Map.Entry<@NotNull CommandSender, @NotNull Shop> data) {
+            return this.executor.apply(data);
+        }
+    }
+
+    public static class Comparator implements java.util.Comparator<Entry> {
```

Comment:
Please give a proper name for this class, for example `ControlPanelEntryComparator`

Reply:
It is the inner class it would look like `ShopControlPanel.Entry` so it's okay